pandas/tests/indexes/datetimes/test_constructors.py

```diff
@@ -787,6 +787,23 @@ def test_construction_with_nat_and_tzlocal(self):
         expected = DatetimeIndex([Timestamp("2018", tz=tz), pd.NaT])
         tm.assert_index_equal(result, expected)
 
+    def test_constructor_with_ambiguous_keyword_arg(self):
+        # GH 35297
+
+        timezone = "America/New_York"
+        start = pd.Timestamp(year=2020, month=11, day=1, hour=1).tz_localize(
```

Comment:
Could you also test where the end is `ambiguous`

Reply:
added

Reply:
added